component/host.go

```diff
@@ -63,4 +64,14 @@ type Host interface {
 	// GetExporters can be called by the component anytime after Component.Start() begins and
 	// until Component.Shutdown() ends.
 	GetExporters() map[config.DataType]map[config.ComponentID]Exporter
+
+	// ReportStatus can be used by a component to communicate its status to registered status
+	// listeners. Components can use this function to indicate that they are functioning properly,
+	// or are in an error state.
+	ReportStatus(eventType status.EventType, componentID config.ComponentID, options ...status.StatusEventOption)
```

Comment:
AFAICT, the `componentID` should be the same on each call that a given component does. Should we therefore remove the `componentID` argument from the public interface? Instead, we could wrap the host we pass to each component so that it sets this automatically.

If we ever need to have a component 'lie' about its component ID, this could still be achieved via an Option.

Reply:
I added the functionality to `hostWrapper`. I had to plumb through the component id by attaching it to the built components. Let me know if there is a better way.

Reply:
This is roughly what I had in mind!